internal/component/discovery/process/process.go

```diff
@@ -44,16 +46,25 @@ type Component struct {
 }
 
 func (c *Component) Run(ctx context.Context) error {
-	doDiscover := func() error {
-		processes, err := discover(c.l, &c.args.DiscoverConfig)
+	var cgroupIDRegexp *regexp.Regexp
+	var err error
+	if c.args.CgroupIDRegex != "" {
```

Comment:
I dont see where updates to the regex are handled in the Update function.

Reply:
You are right. Currently we are not handling regex updates.

I also think it makes more sense to move the regex compilation to `New` function so that we fail fast when provided regular expression is not valid. What do you think?

Question: Can we return error during updates? As we need to verify the regular expression, can we return error for `changed` method when an invalid regular expression is passed during an update?

Reply:
New is fine but we still need to handle updating the regex, we can return an error with update.

Reply:
I pushed the changes to ensure we update regex as well. Let me know what do you think. Cheers